embassy-mspm0/src/wwdt.rs

```diff
@@ -240,6 +240,82 @@ impl ClosedWindowPercentage {
     }
 }
 
+/// Reset cause values from SYSCTL.RSTCAUSE register.
+/// Based on MSPM0 Technical Reference Manual Table 2-9.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[repr(u8)]
+pub enum ResetCause {
+    /// No reset since last read
+    NoReset = 0x00,
+    /// VDD < POR- violation, PMU trim parity fault, or SHUTDNSTOREx parity fault
+    PorHwFailure = 0x01,
+    /// NRST pin reset (>1s)
+    PorExternalNrst = 0x02,
+    /// Software-triggered POR
+    PorSwTriggered = 0x03,
+    /// VDD < BOR- violation
+    BorSupplyFailure = 0x04,
+    /// Wake from SHUTDOWN
+    BorWakeFromShutdown = 0x05,
+    /// Non-PMU trim parity fault
+    BootrstNonPmuParityFault = 0x08,
+    /// Fatal clock fault
+    BootrstClockFault = 0x09,
+    /// NRST pin reset (<1s)
+    BootrstExternalNrst = 0x0C,
+    /// Software-triggered BOOTRST
+    BootrstSwTriggered = 0x0D,
+    /// WWDT0 violation - This is the main watchdog reset cause we're interested in
+    SysrstWwdt0Violation = 0x0E,
+    /// BSL exit (if present)
+    SysrstBslExit = 0x10,
+    /// BSL entry (if present)
+    SysrstBslEntry = 0x11,
+    /// Uncorrectable flash ECC error (if present)
+    SysrstFlashEccError = 0x14,
+    /// CPU lockup violation
+    SysrstCpuLockupViolation = 0x15,
+    /// Debug-triggered SYSRST
+    SysrstDebugTriggered = 0x1A,
+    /// Software-triggered SYSRST
+    SysrstSwTriggered = 0x1B,
+    /// Debug-triggered CPURST
+    CpurstDebugTriggered = 0x1C,
+    /// Software-triggered CPURST
+    CpurstSwTriggered = 0x1D,
+}
+
+impl TryFrom<u8> for ResetCause {
```

Comment:
I don't know if this is public API that we want to guarantee. Since `ResetCause` is a public enum, it would expose `TryFrom` in public API.